tests/validation/test_Steinhardt_Average.py

```diff
@@ -0,0 +1,177 @@
+        freud_data = _compute_comparison_data(gsd_frame, average=False, weighted=True)
+        names = ["q'4", "q'6", "w'4", "w'6"]
+        for i, name in enumerate(names):
+            npt.assert_allclose(
+                rvd_data[:, i], freud_data[:, i], atol=2e-5, err_msg=f"{name} failed"
+            )
+
+    def test_rvd_msm_ave(self):
+        """Check freud against data generated by MSM reference code.
+
+        https://github.com/ArvDee/Minkowski-structure-metrics-calculator
+        """
+        rvd_data = _get_reference_data("RvD_MSM_avq4avq6avw4avw6")
+        gsd_frame = _read_gsd_snapshot("Test_Configuration.gsd")
+        freud_data = _compute_comparison_data(gsd_frame, average=True, weighted=True)
+        names = ["ave. q'4", "ave. q'6", "ave. w'4", "ave. w'6"]
+        for i, name in enumerate(names):
+            # If most of the particles agree, this test is probably correct.
+            # Close investigation showed that numerical issues (truncated
+            # particle positions?) and bonds with facet weights of the order of
+            # 1e-9 caused some deviations in the neighbor counts, which in turn
+            # affected the averaged ql values. This affected 37 out of 3288
+            # particles in the sample data for one of the cases checked.
+            close_values = np.isclose(rvd_data[:, i], freud_data[:, i], atol=5e-5)
+            assert sum(close_values) > (0.985 * len(freud_data)), f"{name} failed"
+
+    @pytest.mark.parametrize(
+        "reference,average,wl",
+        [
+            ("q", False, False),
+            ("w", False, True),
+            ("avq", True, False),
+            ("avw", True, True),
+        ],
+    )
+    def test_msm_calc(self, reference, average, wl):
+        """Check freud against data generated by MSM reference code.
+
+        https://github.com/ArvDee/Minkowski-structure-metrics-calculator
+        """
+        msm_data = _get_reference_data(f"{reference}_0")
+        gsd_frame = _read_gsd_snapshot("Test_Configuration.gsd")
+        lmax = 6
+        freud_data = _compute_msms(gsd_frame, lmax, average, wl)
+        for sph_l in range(lmax + 1):
+            if wl and sph_l == 2:
+                # w'2 tests fail for unknown (probably numerical) reasons.
+                continue
+            name = f"{reference}{sph_l}"
+            # If most of the particles agree, this test is probably correct.
+            # Close investigation showed that numerical issues (truncated
+            # particle positions in the dat file?) and bonds with facet weights
+            # of the order of 1e-9 caused some deviations in the neighbor
+            # counts, which in turn affected the averaged ql values. This
+            # affected 37 out of 3288 particles in the sample data.
+            close_values = np.isclose(
+                msm_data[:, sph_l], freud_data[:, sph_l], atol=1e-5
+            )
+            assert sum(close_values) > (0.985 * len(freud_data)), f"{name} failed"
```

Comment:
```suggestion
            assert np.sum(close_values) > (0.985 * len(freud_data)), f"{name} failed"
```
should be a little faster.